### ISSUE TITLE
Lint/UnneededSplatExpansion and Lint/PercentStringArray cause nested single/double quotes to error mid-lint

### ISSUE THREAD

**Issue by tadiou** [rubocop-hq/rubocop#6506]
## When

During Autocorrecting it doesn't properly escape nested quotes in a percent string array.

## Expected behavior

```ruby
execute(:bundle, *%w(config --local build.sigar "--with-cppflags='-fgnu89-inline'"))
```
or
```ruby
execute(:bundle, "config", "--local", "build.sigar", "--with-cppflags='-fgnu89-inline'")
```

## Actual behavior

```ruby
execute(:bundle, 'config', '--local', 'build.sigar', '"--with-cppflags='-fgnu89-inline'"')
```

## Other Notes

We're using
```yaml
Style/StringLiterals:
  EnforcedStyle: double_quotes
```

It enters a state where it can't continue because there's a quote imbalance now, and fails to correct the `Style/StringLiteral` to properly enforce the double quote.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```
